src/ethernet_interface.cpp

```diff
@@ -427,6 +427,44 @@ void EthernetInterface::addStaticGateway(const StaticGatewayInfo& info)
     }
 }
 
+bool EthernetInterface::dhcpIsEnabled(IP::Protocol family, bool ignoreProtocol)
+{
+    return ((EthernetInterfaceIntf::dhcpEnabled() ==
+             EthernetInterface::DHCPConf::both) ||
+            ((EthernetInterfaceIntf::dhcpEnabled() ==
+              EthernetInterface::DHCPConf::v6) &&
```

Comment:
Should you also check for v4v6stateless -> this also means dhcpv4 is enabled right?

Reply:
done